### PR TITLE
Restore compatibility with Gradle's configuration cache

### LINKED ISSUE #95: apiDump task incompatible with configuration cache
```
FAILURE: Build failed with an exception.

* What went wrong:
Configuration cache problems found in this build.

10 problems were found storing the configuration cache.
- Task `:detekt-api:apiDump` of type `org.gradle.api.tasks.Sync`: cannot serialize object of type 'kotlinx.validation.KotlinApiBuildTask', a subtype of 'org.gradle.api.Task', as these are not supported with the configuration cache.
  See https://docs.gradle.org/7.5/userguide/configuration_cache.html#config_cache:requirements:task_access
- Task `:detekt-api:apiDump` of type `org.gradle.api.tasks.Sync`: cannot serialize object of type 'org.gradle.api.internal.project.DefaultProject', a subtype of 'org.gradle.api.Project', as these are not supported with the configuration cache.
  See https://docs.gradle.org/7.5/userguide/configuration_cache.html#config_cache:requirements:disallowed_types
```

### PR DESCRIPTION
**Pull request by 3flex**
Fixes #95

### CODE REVIEW

File: src/main/kotlin/BinaryCompatibilityValidatorPlugin.kt [hunk 1]
```diff
@@ -261,15 +261,12 @@ private fun Project.configureCheckTasks(
...
 
     val apiDump = task<Sync>(targetConfig.apiTaskName("Dump")) {
-        isEnabled = apiCheckEnabled(extension) && apiBuild.map { it.enabled }.getOrElse(true)
+        isEnabled = apiCheckEnabled(projectName, extension) && apiBuild.map { it.enabled }.getOrElse(true)
         group = "other"
         description = "Syncs API from build dir to ${targetConfig.apiDir} dir for $projectName"
         from(apiBuildDir)
         into(apiCheckDir)
         dependsOn(apiBuild)
-        doFirst {
```
Comment (qwwdfsad) [inline src/main/kotlin/BinaryCompatibilityValidatorPlugin.kt:270]: Could you please elaborate on why you removed this `mkdirs` call?

Does `into` documentation guarantees `mkdir`?

**Review by qwwdfsad — COMMENTED** [review]
Thanks for taking care of it!

I have a question about `mkdirs` and this PR is good to go

File: src/main/kotlin/BinaryCompatibilityValidatorPlugin.kt (hunk 1 above)
Comment (3flex) [reply src/main/kotlin/BinaryCompatibilityValidatorPlugin.kt:270]: It's odd, as this doesn't seem to violate the configuration cache contract, but it wouldn't work unless I removed it.

In any case, yes, `into` will create the required directories. More here: https://docs.gradle.org/current/userguide/working_with_files.html#sec:creating_directories_example

> All core Gradle tasks ensure that any output directories they need are created if necessary

Comment (qwwdfsad) [reply src/main/kotlin/BinaryCompatibilityValidatorPlugin.kt:270]: Works for me, thanks!
At least now we have a proper test coverage for that

**Review by qwwdfsad — APPROVED** [review]
Once again, thanks for taking care of it!

**Comment by 3flex** [comment]
Any chance we could get a new release cut? Not sure if there's anything else planned for the next version, but I'd really like to get rid of the `notCompatibleWithConfigurationCache` call for these tasks in the build scripts!

**Comment by qwwdfsad** [comment]
0.11.1 is published to maven central. Should be available for regular dependencies within an hour

**Pull request merged**